src/plays/quote-generator/QuoteGenerator.jsx

```diff
@@ -0,0 +1,138 @@
+import { getPlayById } from "meta/play-meta-util";
+import PlayHeader from "common/playlists/PlayHeader";
+import "./quoteGenerator.css";
+import { useEffect, useState } from "react";
+
+const fetchQuote = () => {
+  const response = fetch("https://api.quotable.io/random");
+  return response;
+};
+
+function QuoteGenerator(props) {
+  // Do not remove the below lines.
+  // The following code is to fetch the current play from the URL
+  const { id } = props;
+  const play = getPlayById(id);
+
+  // Your Code Start below.
+  const [quoteArray, setQuoteArray] = useState([]);
+  const [current, setCurrent] = useState(0);
+
+  const updateState = (newState, stateUpdator) => {
+    return new Promise((resolve) => {
+      return stateUpdator(newState, resolve());
+    });
+  };
+
+  const previousQuote = async () => {
+    await updateState(current - 1, setCurrent);
+    return;
+  };
+
+  const nextQuote = async () => {
+    if (current < quoteArray.length - 1) {
+      await updateState(current + 1, setCurrent);
+      return;
+    }
+
+    const response = await (await fetchQuote()).json();
+    await updateState(
+      [...quoteArray, [response.content, response.author]],
+      setQuoteArray
+    );
+    await updateState(current + 1, setCurrent);
+    return;
+  };
+
+  useEffect(() => {
+    const fetcher = async () => {
+      const response = await (await fetchQuote()).json();
+      await updateState(
+        [...quoteArray, [response.content, response.author]],
+        setQuoteArray
+      );
+    };
+    fetcher();
+  }, []);
+
+  return (
+    <>
+      <div className="play-details">
+        <PlayHeader play={play} />
+        <div className="play-details-body">
```

Comment:
Add a class here like 'rand-qutote-gen' and use the same class to proceed your style in the CSS file.

Reply:
Changes Added

Reply:
No what I meant here is..

Do like this,

```html
<div className="play-details-body rand-qutote-gen">
````

now in your CSS file prefix all the styles using `.rand-qutote-gen `. It will make sure, your styles will never get a conflict with others/

Reply:
Yes I am asking to add another unique class name and use that as a prefix to all your styles.

For example check this play:

- Added a unique class name: https://github.com/atapas/react-play/blob/a8358f38534b6981e18b98d55e6fc7ca842e9037/src/plays/memory-game/NetlifyCardGame.jsx#L202

- Used the same unique class name as prefix for all styles: https://github.com/atapas/react-play/blob/a8358f38534b6981e18b98d55e6fc7ca842e9037/src/plays/memory-game/NetlifyCardGame.scss#L9

Let me know if you have questions.

Reply:
Basically create a hierarchy starting with the unique class name I provide to style the exact desired element?

Reply:
That's right and safer.

Reply:
is it called specificity ?

Reply:
Right, one part of CSS specificity.

Reply:
@Abhishek-90 Let me know when it is done. I'll do a final review.

Reply:
Changes added.